octobot/src/server/github_handler.rs

```diff
@@ -58,6 +59,50 @@ pub struct GithubEventHandler {
     pub pr_merge: Arc<dyn Worker<PRMergeRequest>>,
     pub repo_version: Arc<dyn Worker<RepoVersionRequest>>,
     pub force_push: Arc<dyn Worker<ForcePushRequest>>,
+    pub team_members_cache: TeamsCache,
+}
+
+struct TeamCacheEntry {
+    users: Vec<github::User>,
+    expiry: Instant,
+}
+
+pub struct TeamsCache {
+    members: Mutex<collections::HashMap<String, TeamCacheEntry>>,
+    ttl: Duration,
+}
+
+impl TeamsCache {
+    pub fn new(ttl: Duration) -> TeamsCache {
+        TeamsCache {
+            members: Mutex::new(collections::HashMap::new()),
+            ttl,
+        }
+    }
+
+    pub fn insert(&self, team_slug: String, users: Vec<github::User>) {
+        let mut hash = self.members.lock().unwrap();
+        let entry = TeamCacheEntry {
+            users,
+            expiry: Instant::now().add(self.ttl),
+        };
+        hash.insert(team_slug, entry);
+    }
+
+    pub fn get(&self, team_slug: &String) -> Option<Vec<github::User>> {
+        let hash = self.members.lock().unwrap();
```

Comment:
```suggestion
        let mut hash = self.members.lock().unwrap();
```

grabbing the lock again with this lock already held online 100 is going to deadlock i'd think.

should just mutably grab the lock here.